### PR TITLE
Add rpc format doc

### LINKED ISSUE #119: CBOR encoding specification for IDL types
All runtime clients need to implement this specification to send properly serialized requests to the runtime.

### PR DESCRIPTION
**Pull request by nhynes**
low-effort resolves #119

### CODE REVIEW

File: docs/rpc_format.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Oasis RPC Format
+
+This doc describes the message format expected by services generated using oasis-rs.
+This is the format used by [oasis.js](https://github.com/oasislabs/oasis.js) when invoking methods on service objects loaded from an IDL.
+
+The types are defined in the [oasis-rpc](/oasis-rpc/src/lib.rs) crate and the wire format is [CBOR](https://tools.ietf.org/html/rfc7049).
+You can find an example of the interface definition language (IDL) in the [tests](/tests/idl-gen/res/TestService.json).
+
+Note: If you find that CBOR is not efficient enough, or don't need struct-as-state semantics, you can build your service using `oasis build --wasi`, which will skip generating an interface or method dispatch tree.
```
Comment (armaniferrante) [inline docs/rpc_format.md]: TODO: We should link to a doc/example doing this. If none exists, let's file an issue. Low priority (obviously).

File: docs/rpc_format.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+* `Enum` is a finite set of strings
+* `Event` is mostly a struct, but one that's expected to be used as a template for recording events on the blockchain.
+  Unlike a struct, `Event` can have up to three `indexed` fields, which tell blockchain indexers (e.g.,  gateways) to make these fields amenable to fast lookup.
+
+An interface definition will link to other interface definitions that it depends on for externally defined types.
+
+A note on defined types: you can only expose types in an RPC interface that were (transitively) also defined in an RPC interface.
+This is to say that you can't just return a type from a crate--even if it's serializable.
+This is because non-*your preferred language* clients and services aren't able to pull down libraries written in other languages.
+The quick solution is to re-define the type in your own interface, but if you find this annoying, you should file an issue, so that we know to work on automating this step.
```
Comment (armaniferrante) [inline docs/rpc_format.md]: We should have an existing issue that we link to here instead of encouraging new issues?

File: docs/rpc_format.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+### Functions
+
+Functions can be either methods on an deployed service or the `Constructor`.
+A `Constructor` is an anonymous function that takes inputs and, optionally, returns an error; it does not return anything because the output is the service that's persisted to the blockchain.
+Otherwise, a `Function` is named, has named arguments (`Field`s), and can return whatever (as long as it's an RPC `Type`).
+A `Function` can be marked as mutable or immutable, but this isn't enforced by the platform; it's just to help the author of the service write safer code.
+
+## Wire format
+
+The wire format is standard CBOR; nothing fancy.
```
Comment (armaniferrante) [inline docs/rpc_format.md]: We should be more specific. I.e., that the wire format is

```
cbor.encode({
      method: fn.name,
      payload: args, // array
    });
```

File: docs/rpc_format.md (hunk 1 above)
Comment (nhynes) [reply docs/rpc_format.md]: https://github.com/oasislabs/tutorials/issues/33

**Pull request merged**